benchexec/tools/template.py

```diff
@@ -22,17 +35,386 @@ class UnsupportedFeatureException(benchexec.BenchExecException):
     pass
 
 
-class BaseTool(object):
+class BaseTool2(object, metaclass=ABCMeta):
     """
     This class serves both as a template for tool-info implementations,
     and as an abstract super class for them.
     For writing a new tool info, inherit from this class and override
     the necessary methods (always executable() and name(),
     usually determine_result(), cmdline(), and version(),
     and maybe working_directory() and get_value_from_output(), too).
-    The class for each specific tool need to be named "Tool".
-    For more information, please refer to
-    https://github.com/sosy-lab/benchexec/blob/master/doc/tool-integration.md
+
+    In special circumstances, it can make sense to not inherit from this class.
+    In such cases the tool-info module's class needs to implement all the methods
+    defined here and BaseTool2.register needs to be called with the respective class
+    (cf. documentation of abc.ABCMeta.register) to declare compatibility with BaseTool2.
```

Comment:
This would be fine, and also calling `register(Tool)` afterwards in the module source. I prefer not to add too much details here, though, in order to not confuse and distract from the regular method. This info is also not specific to tool-info modules, it is standard `ABCMeta` stuff.